server/src/main/java/org/elasticsearch/cluster/metadata/DesiredNode.java

```diff
@@ -354,42 +382,42 @@ public String toString() {
             + '}';
     }
 
-    private static boolean invalidNumberOfProcessors(float processors) {
-        return processors <= 0 || Float.isInfinite(processors) || Float.isNaN(processors);
-    }
-
-    private static int roundUp(float value) {
-        return (int) Math.ceil(value);
-    }
-
-    private static int roundDown(float value) {
-        return Math.max(1, (int) Math.floor(value));
+    private static boolean invalidNumberOfProcessors(Processors processors) {
+        return processors != null && processors.count() <= 0;
```

Comment:
I think this can only be true for `processors.count() == 0` now, but is that even a legal case for `Processors`? Given that we now validate `Processors` separately perhaps this method is no longer needed?